File: 0242-valid-anagram/0242-valid-anagram.c
#include <stdbool.h>
#include <string.h>

bool isAnagram(char* s, char* t) {
    int sLen = strlen(s);
    int tLen = strlen(t);
    if (sLen != tLen) return false;

    int charCounts[26] = {0};

    for (int i = 0; i < sLen; i++) {
        charCounts[s[i] - 'a']++;
        charCounts[t[i] - 'a']--;
    }

    for (int i = 0; i < 26; i++) {
        if (charCounts[i] != 0) return false;
    }

    return true;
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
            
        # Using a fixed 26-slot list is faster than a hash dictionary 
        # because it avoids string hashing calculation cycles.
        char_counts = [0] * 26
        
        for i in range(len(s)):
            char_counts[ord(s[i]) - 97] += 1
            char_counts[ord(t[i]) - 97] -= 1
            
        for count in char_counts:
            if count != 0:
                return False
                
        return True
